src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts

```diff
@@ -153,39 +152,39 @@ describe('CourseExerciseDetailsComponent', () => {
                 fixture = TestBed.createComponent(CourseExerciseDetailsComponent);
                 comp = fixture.componentInstance;
 
-                // stub profileService
+                // mock profileService
                 profileService = fixture.debugElement.injector.get(ProfileService);
-                getProfileInfoStub = stub(profileService, 'getProfileInfo');
+                getProfileInfoMock = jest.spyOn(profileService, 'getProfileInfo');
                 const profileInfo = { inProduction: false } as ProfileInfo;
                 const profileInfoSubject = new BehaviorSubject<ProfileInfo | null>(profileInfo);
-                getProfileInfoStub.returns(profileInfoSubject);
+                getProfileInfoMock.mockReturnValue(profileInfoSubject);
 
-                // stub exerciseService
+                // mock exerciseService
                 exerciseService = fixture.debugElement.injector.get(ExerciseService);
-                getExerciseDetailsStub = stub(exerciseService, 'getExerciseDetails');
-                getExerciseDetailsStub.returns(of({ body: exercise }));
+                getExerciseDetailsMock = jest.spyOn(exerciseService, 'getExerciseDetails');
+                getExerciseDetailsMock.mockReturnValue(of({ body: exercise }));
 
-                // stub teamService, needed for team assignment
+                // mock teamService, needed for team assignment
                 teamService = fixture.debugElement.injector.get(TeamService);
-                getTeamPayloadStub = stub(teamService, 'teamAssignmentUpdates');
+                getTeamPayloadMock = jest.spyOn(teamService, 'teamAssignmentUpdates', 'get');
                 const teamAssignmentPayload = { exerciseId: 2, teamId: 2, studentParticipations: [] } as TeamAssignmentPayload;
-                getTeamPayloadStub.get(() => Promise.resolve(of(teamAssignmentPayload)));
+                getTeamPayloadMock.mockReturnValue(() => Promise.resolve(of(teamAssignmentPayload)));
 
-                // stub participationService, needed for team assignment
+                // mock participationService, needed for team assignment
                 participationService = fixture.debugElement.injector.get(ParticipationService);
-                mergeStudentParticipationStub = stub(participationService, 'mergeStudentParticipations');
+                mergeStudentParticipationMock = jest.spyOn(participationService, 'mergeStudentParticipations');
```

Comment:
Since this one doesn't get defined as a Mock here I suggest moving this down to the places where it gets used. It's only one line so that shouldn't be an issue.